src/util/projectDelete.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactNode } from "react";
+import { filterUsedByType } from "./usedBy";
+import { ResourceType } from "./resourceDetails";
+import { LxdProject } from "types/project";
+import { Link } from "react-router-dom";
+
+export const generateProjectUsedByTooltip = (project: LxdProject) => {
+  const resourceLabelAndLink: Record<string, { label: string; link: string }> =
+    {
+      instance: {
+        label: "Instances",
+        link: `/ui/project/${project.name}/instances`,
+      },
+      profile: {
+        label: "Profiles",
+        link: `/ui/project/${project.name}/profiles`,
+      },
+      image: {
+        label: "Images",
+        link: `/ui/project/${project.name}/images`,
+      },
+      volume: {
+        label: "Custom volumes",
+        link: `/ui/project/${project.name}/storage/volumes`,
+      },
+    };
+
+  const resourceTypes = Object.keys(resourceLabelAndLink);
+  const usedByItems: ReactNode[] = [];
+  for (const resourceType of resourceTypes) {
+    const usedBy = filterUsedByType(
+      resourceType as ResourceType,
+      project.used_by,
+    );
+
+    if (usedBy.length > 0) {
+      const label = resourceLabelAndLink[resourceType].label;
+      const link = resourceLabelAndLink[resourceType].link;
+      usedByItems.push(
+        <p key={resourceType} className="is-dark u-no-margin--bottom">
+          {link ? <Link to={link}>{label}</Link> : label} ({usedBy.length})
+        </p>,
+      );
+    }
+  }
+
+  return (
+    <aside>
```

Comment:
```suggestion
    <>
```
Not sure if aside is the right thing to use here. Haven't seen it used for the tooltip message before. Maybe just a plain fragment works as well?